reddit_service_activity/__init__.py

```diff
@@ -59,18 +59,45 @@ def record_activity(self, context, context_id, visitor_id):
         self.counter.record_activity(context.redis, context_id, visitor_id)
 
     def count_activity(self, context, context_id):
-        if not _ID_RE.match(context_id):
+        results = self.count_activity_multi(context, [context_id])
+        return results[context_id]
+
+    def count_activity_multi(self, context, context_ids):
+        if not all(_ID_RE.match(context_id) for context_id in context_ids):
             raise ActivityService.InvalidContextIDException
 
-        cache_key = "{context}/cached".format(context=context_id)
-        cached_result = context.redis.get(cache_key)
-        if cached_result:
-            return ActivityInfo.from_json(cached_result.decode())
-        else:
-            count = self.counter.count_activity(context.redis, context_id)
-            info = ActivityInfo.from_count(self.fuzz_threshold, count)
-            context.redis.setex(cache_key, _CACHE_TIME, info.to_json())
-            return info
+        activity = {}
+
+        cache_keys = [context_id + "/cached" for context_id in context_ids]
```

Comment:
It may be super obvious what's going on here and I'm just missing it, but simple header-esque comments would be nice to explain the stuff that is logically grouped together.